### PR TITLE
docs: update CLI for time travel connection and list shared projects

### PR DESCRIPTION
**Pull request by bgrenon**
Updates for time travel connections:
https://neon-next-git-bgrenon-cli-timetravel-sharedproject-neondatabase.vercel.app/docs/guides/time-travel-assist
https://neon-next-git-bgrenon-cli-timetravel-sharedproject-neondatabase.vercel.app/docs/reference/cli-connection-string

Updates for listing shared projects:
https://neon-next-git-bgrenon-cli-timetravel-sharedproject-neondatabase.vercel.app/docs/reference/cli-projects#list
https://neon-next-git-bgrenon-cli-timetravel-sharedproject-neondatabase.vercel.app/docs/guides/branching-neon-cli#list-branches-with-the-cli

@stepashka Missing sections are now added to the Branching with CLI page:
- [Create branch from non-primary parent](https://neon-next-git-bgrenon-cli-timetravel-sharedproject-neondatabase.vercel.app/docs/guides/branching-neon-cli#create-a-branch-from-a-non-primary-parent)
- [Reset from parent](https://neon-next-git-bgrenon-cli-timetravel-sharedproject-neondatabase.vercel.app/docs/guides/branching-neon-cli#resetting-a-branch-to-its-parent)

### CODE REVIEW

File: content/docs/reference/cli-connection-string.md [hunk 1]
```diff
@@ -19,10 +19,10 @@ This command gets a Postgres connection string for connecting to a database in y
 ### Usage
 
 ```bash
-neonctl connection-string [branch] [options]
+neonctl connection-string [branch[@timestamp|@LSN]] [options]
 ```
 
-`branch` specifies the branch name or id. If a branch name or ID is ommited, the primary branch is used.
+`branch` specifies the branch name or id. If a branch name or ID is omitted, the primary branch is used. `@timestamp|@LSN` is used to specify a specific point in the branch's history for time travel connections. If omitted, current (HEAD) is used.
```
Comment (danieltprice) [inline content/docs/reference/cli-connection-string.md:25]:
```suggestion
`branch` specifies the branch name or ID. If a branch name or ID is omitted, the primary branch is used. `@timestamp|@LSN` is used to specify a specific point in the branch's history for time travel connections. If omitted, the current state (HEAD) is used.
```

File: content/docs/reference/cli-connection-string.md [hunk 2]
```diff
@@ -63,6 +63,12 @@ In addition to the Neon CLI [global options](/docs/reference/neon-cli#global-opt
    postgres://alex:AbC123dEf@ep-cool-darkness-123456.us-east-2.aws.neon.tech/dbname?connect_timeout=30
     ```
 
+- Get a connection string to a specific point in a branch's history by appending `@timestamp` or `@lsn`. Availability depends on your configured [history retention](/docs/manage/projects#configure-history-retention) window.
+
+    ```bash
+    neonctl connection-string @2024-04-21T00:00:00Z
+    ```
+
```
Comment (danieltprice) [inline content/docs/reference/cli-connection-string.md:71]: For additional examples, see [How to use Time Travel](/docs/guides/time-travel-assist#how-to-use-time-travel).

Comment (danieltprice) [inline content/docs/reference/cli-connection-string.md:71]:
```suggestion
    For additional examples, see [How to use Time Travel](/docs/guides/time-travel-assist#how-to-use-time-travel).
```

**Review by danieltprice — APPROVED** [review]
LGTM! Just a few minor comments.

File: content/docs/guides/branching-neon-cli.md [hunk 3]
```diff
@@ -50,19 +50,56 @@ connection_uris
 The Neon CLI provides a `neonctl connection-string` command you can use to extract a connection uri programmatically. See [Neon CLI commands — connection-string](https://neon.tech/docs/reference/cli-connection-string).
 </Admonition>
 
+## Create a branch from a non-primary parent
```
Comment (duskpoet) [inline content/docs/guides/branching-neon-cli.md:53]: this stuff doesn't look like it is related to the topic of pr, is it an intentional change?

**Review by duskpoet — APPROVED** [review]
Great, thanks!

File: content/docs/guides/branching-neon-cli.md (hunk 3 above)
Comment (bgrenon) [reply content/docs/guides/branching-neon-cli.md:53]: You're right. It was some missing content that we added while in these files.

**Pull request merged**